packages/shorebird_cli/lib/src/executables/shorebird_tool.dart

```diff
@@ -0,0 +1,83 @@
+import 'dart:io';
+
+import 'package:mason_logger/mason_logger.dart';
+import 'package:path/path.dart' as p;
+import 'package:scoped/scoped.dart';
+import 'package:shorebird_cli/src/shorebird_env.dart';
+import 'package:shorebird_cli/src/shorebird_process.dart';
+
+/// A reference to a [ShorebirdTool] instance.
+final shorebirdToolRef = create(ShorebirdTool.new);
+
+/// The [ShorebirdTool] instance available in the current zone.
+ShorebirdTool get shorebirdTool => read(shorebirdToolRef);
+
+/// {@template package_failed_exception}
+/// An exception thrown when a package fails.
+/// {@endtemplate}
+class PackageFailedException implements Exception {
+  /// {@macro package_failed_exception}
+  PackageFailedException(this.message);
+
+  final String message;
+
+  @override
+  String toString() => message;
+}
+
+/// A wrapper around the `shorebird_tools` executable.
+///
+/// Used to access many commands related to Shorebird's flutter tooling.
+class ShorebirdTool {
```

Comment:
```suggestion
class ShorebirdTools {
```
can we rename for consistency with the exectuable?